### PR TITLE
17 BE: Thread and Post models created

### PR DESCRIPTION
**Pull request by ashan36**
Also changed the user model to make use of mongoose's built in timestamps

### CODE REVIEW

**Review by nicholasrobinharrison — APPROVED** [review]
looks great

File: server/database/models/Post.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { Schema, model } = require("mongoose");
+
+var postSchema = new Schema(
+  {
+    u_id: {
```
Comment (skassam21) [inline server/database/models/Post.js:6]: This should be an objectid

File: server/database/models/Thread.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const { Schema, model } = require("mongoose");
+const { postSchema } = require("./Post");
+
+var threadSchema = new Schema(
+  {
+    creator_id: {
```
Comment (skassam21) [inline server/database/models/Thread.js:6]: should be object Id as well?

File: server/database/models/Post.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const { Schema, model } = require("mongoose");
+
+var postSchema = new Schema(
```
Comment (skassam21) [inline server/database/models/Post.js]: random sidenotes, I believe we should be using const over var for these

File: server/database/models/User.js [hunk 4]
```diff
@@ -2,28 +2,27 @@ const { Schema, model } = require("mongoose");
...
-  },
-  password: {
-    type: String,
-    required: true
-  },
-  experience: {
-    type: Schema.Types.Mixed,
-    of: Number
+var userSchema = new Schema(
+  {
```
Comment (skassam21) [inline server/database/models/User.js:7]: Something changed with the formatting here, make sure we have all the same prettier.io settings on back-end too

File: server/database/models/Post.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  {
+    u_id: {
+      type: String,
+      required: true
+    },
+    tite: {
```
Comment (skassam21) [inline server/database/models/Post.js]: spelling mistake

File: server/database/models/Thread.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+      type: String,
+      required: true
+    },
+    status: {
+      type: Number,
+      default: 0,
+      required: true
+    },
+    reviewer_id: String,
```
Comment (skassam21) [inline server/database/models/Thread.js:15]: object ID?

**Pull request merged**